packages/supabase_flutter/lib/src/supabase_auth.dart

```diff
@@ -107,33 +108,15 @@ class SupabaseAuth with WidgetsBindingObserver {
   void didChangeAppLifecycleState(AppLifecycleState state) {
     switch (state) {
       case AppLifecycleState.resumed:
-        _recoverSupabaseSession();
+        Supabase.instance.client.auth.startAutoRefresh();
+      case AppLifecycleState.detached:
+      case AppLifecycleState.inactive:
+      case AppLifecycleState.paused:
+        Supabase.instance.client.auth.stopAutoRefresh();
```

Comment:
What's the reason to stop auth refresh here? From what I read in the [documentation](https://api.flutter.dev/flutter/dart-ui/AppLifecycleState.html) the app may still be running in the background.

Reply:
Oh sorry, seems like I didn't read that.
But does it matter if a refresh request fails? The app could still be running, so the application could send requests to supabase with outdated access token. If the refresh fails, then requests from the application will also fail, which is fine.

Reply:
No problem at all!

>The app could still be running, so the application could send requests to supabase with outdated access token. If the refresh fails, then requests from the application will also fail, which is fine.

Actually, `stopAutoRefresh()` only stops periodically checking the token if it's expired then refresh if expired, and if a user would to make a request after `stopAutoRefresh()` has been called, the auth HTTP client will still refresh the token as needed. 

All it's doing is avoiding a slim chance of the auth refresh request firing, which reaches the server, but before the response comes back the connection is lost. This is where the app could be left with an expired JWT. This could very much happen when the app is in the foreground as well, but if the app is running for a long time in the background, there might be more chances of something like this happening. 

It was advice from our Auth engineer and we also recommend it for React native apps. On step 4 there is pretty much the same code:
https://supabase.com/docs/guides/auth/quickstarts/react-native

Reply:
Oh yeah I forgot that the auth http client still exists. Then you are right this is fine.